pyccel/parser/semantic.py

```diff
@@ -355,6 +355,47 @@ def annotate(self):
     #              Utility functions for scope handling
     #================================================================
 
+    def create_new_function_scope(self, syntactic_name, semantic_name, **kwargs):
+        """
+        Create a new Scope object for a Python function.
+
+        Create a new Scope object for a Python function with the given name,
+        and attach any decorators' information to the scope. The new scope is
+        a child of the current one, and can be accessed from the dictionary of
+        its children using the function name as key.
+
+        Before returning control to the caller, the current scope (stored in
+        self._scope) is changed to the one just created, and the function's
+        name is stored in self._current_function_name.
+
+        Parameters
+        ----------
+        syntactic_name : str
+            Function's original name in the translated code, used as a key to
+            retrieve the new scope.
+
+        semantic_name : str
+            The new name of the function by which it will be known in the target
+            language.
+
+        **kwargs : dict
+            Keyword arguments passed through to the new scope.
+
+        Returns
+        -------
+        Scope
+            The new scope for the function.
+        """
+        child = self.scope.new_child_scope(syntactic_name, **kwargs)
+        child.local_used_symbols[syntactic_name] = semantic_name
+        child.python_names[semantic_name] = syntactic_name
+
+
+        self._scope = child
```

Comment:
One blank line should be enough here

Reply:
Fixed [c024b87](https://github.com/pyccel/pyccel/pull/2314/commits/c024b8736cd968746fa234c412d99cdf4d42133f)

Reply:
Maybe we should think about using a tool such as [`black`](https://black.readthedocs.io/en/stable/index.html) to ensure consistent formatting?

Reply:
Yes! Good idea!